tests/ut/azure_iot_jws_mbedtls_ut.c

```diff
@@ -0,0 +1,181 @@
+                                      "kVoQmVLclh6YkViV2tFNm4zTEoxbkd5M1htUlVFcER0Umdpa0tBUzZybFhFT0VneXNjIn0.eyJzaGEyNTYiOmFiY2pPWEdzVm9DV1ozQ3V0WGxZY1dDZ"
+                                      "VhjZXdiRHhGL0ZuNWozaytJbVk9In0.Wq4UoXt4dGay_P8uy7jrxM8Iip3KCXkGZvQwnu83704CzDogfVqX4GegT68s47veOi3x2Gf5rjX7vOMzVf9Ck0"
+                                      "ylGCfon-vit938hO9MNYM7siA5htYHzotdECD1LfI_BjlLxkwXt0OyLC1PJvMw9N870pb51NtTon0OmaQslEyf6ih6DrEvsNUnyjRcrzSWlIyRo18kqlz"
+                                      "eetARTYE7qGQr7oZPh0RWXVP5b5XR3wbJ_IeZ6i85YmjFpbRGJaSPCuzpa7XKvvFzB5rB5lGmbkWsOMyLbVzUriW87BzbB06g-wzs1S-z07s-ZGjTbFdr"
+                                      "XrGjkKtv3TaDirjTqHhhJyI2cVLBctr4Wv4XITPyZeJt2KcIQZup-KfCRNbM3c3_PXPgvJtOg5BhmUrUKGMqFTl84EIB44B1QqKmuiTdH3bNQxPKBecpC"
+                                      "k-O9g03pB-fk1D_3sL1ju364STs87s77DfGK9e0oHbHgfzp4EdgrwRQBvTCWWKG3iT6ByfSH4N0";
+static char ucScratchBuffer[ azureiotjwsSCRATCH_BUFFER_SIZE ];
+
+static int prvInitMbedTLS( mbedtls_entropy_context * pxEntropyContext,
+                           mbedtls_ctr_drbg_context * pxCtrDrgbContext )
+{
+    int32_t lMbedtlsError = 0;
+
+    /* Set the mutex functions for mbed TLS thread safety. */
+    mbedtls_threading_set_alt( mbedtls_platform_mutex_init,
+                               mbedtls_platform_mutex_free,
+                               mbedtls_platform_mutex_lock,
+                               mbedtls_platform_mutex_unlock );
+
+    /* Initialize contexts for random number generation. */
+    mbedtls_entropy_init( pxEntropyContext );
+    mbedtls_ctr_drbg_init( pxCtrDrgbContext );
+
+    /* Add a strong entropy source. At least one is required. */
+    lMbedtlsError = mbedtls_entropy_add_source( pxEntropyContext,
+                                                mbedtls_platform_entropy_poll,
+                                                NULL,
+                                                32,
+                                                MBEDTLS_ENTROPY_SOURCE_STRONG );
+
+    if( lMbedtlsError != 0 )
+    {
+        printf( "mbedtls_entropy_add_source failed: %i\n", lMbedtlsError );
+        return lMbedtlsError;
+    }
+
+    /* Seed the random number generator. */
+    lMbedtlsError = mbedtls_ctr_drbg_seed( pxCtrDrgbContext,
+                                           mbedtls_entropy_func,
+                                           pxEntropyContext,
+                                           "jws_ut",
+                                           strlen( "jws_ut" ) );
+
+    if( lMbedtlsError != 0 )
+    {
+        printf( "mbedtls_ctr_drbg_seed failed: %i\n", lMbedtlsError );
+    }
+
+    return lMbedtlsError;
+}
+
+static int setup( void ** state )
+{
+    memset( ucScratchBuffer, 0, sizeof( ucScratchBuffer ) );
+    return 0;
+}
+
+static void testAzureIoTJWS_ManifestAuthenticate_Success( void ** ppvState )
+{
+    assert_int_equal( AzureIoTJWS_ManifestAuthenticate( ucValidManifest, strlen( ucValidManifest ),
```

Comment:
I think the expected value should be the first parameter. 
Some UT frameworks will differentiate and the error message when the test fails may be confusing. Even if cmocka doesn't, please take it as a nit (for code readability).

Reply:
I hear you. Unfortunately most of the tests (if not all) are this way:

![image](https://user-images.githubusercontent.com/36710865/194578273-5917951c-52b7-48cb-ab12-755b973e39cf.png)